### PR TITLE
Addon-docs: Support `<StrictMode />` and `<Suspense />` in source viewer

### LINKED ISSUE #22472: [Bug]: React.StrictMode is not working
### Describe the bug

`strictMode: true` option is ignored when using `@storybook/react-webpack5` in v. 7.0.*

### To Reproduce

The bug is reproducible in '[React v18 Webpack (TypeScript)](https://stackblitz.com/edit/github-jzuuqp?file=.storybook%2Fmain.ts)' on StackBlitz.

Or locally:

1. `npm init -y`
2. `npx storybook@latest init`
3. Choose `webpack_react` project type
4. Yes to `missing-babelrc` migration
5. Set `framework.options.strictMode` in `main.js` to `true`
6. `npm run storybook`

### System

```shell
Environment Info:

  System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
  Binaries:
    Node: 16.19.0 - C:\Program Files\nodejs\node.EXE       
    Yarn: 3.4.1 - C:\Program Files\nodejs\yarn.CMD
    npm: 8.19.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (113.0.1774.35)
  npmPackages:
    @storybook/addon-essentials: ^7.0.9 => 7.0.9
    @storybook/addon-interactions: ^7.0.9 => 7.0.9
    @storybook/addon-links: ^7.0.9 => 7.0.9
    @storybook/blocks: ^7.0.9 => 7.0.9
    @storybook/react: ^7.0.9 => 7.0.9
    @storybook/react-webpack5: ^7.0.9 => 7.0.9
    @storybook/testing-library: ^0.0.14-next.2 => 0.0.14-next.2
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zhyd1997**
Issue: #11554

## What I did

Refactor:

https://github.com/storybookjs/storybook/blob/9a07fcba8a581558d0fa8262989ef83b70a507a0/code/renderers/react/src/docs/jsxDecorator.tsx#L77-L92

So that we will not write redundant code here: 

https://github.com/algolia/react-element-to-jsx-string/blob/984f26908c17cea134fa53d17ac51f179e5195b2/src/parser/parseReactElement.js#L47-L77

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/renderers/react/src/docs/jsxDecorator.tsx [hunk 1]
```diff
@@ -74,21 +72,16 @@ export const renderJsx = (code: React.ReactElement, options: JSXOptions) => {
...
-            el.type.displayName ||
-            (el.type === Symbol.for('react.profiler') ? 'Profiler' : null) ||
-            getDocgenSection(el.type, 'displayName') ||
-            (el.type.name !== '_default' ? el.type.name : null) ||
-            (typeof el.type === 'function' ? 'No Display Name' : null) ||
-            (isForwardRef(el.type) ? el.type.render.name : null) ||
-            (isMemo(el.type) ? el.type.type.name : null) ||
-            el.type,
-        };
+  let displayNameDefaults;
```
Comment (zhyd1997) [inline code/renderers/react/src/docs/jsxDecorator.tsx:75]: **Major Change**: `displayNameDefaults.displayName` should be `undefined` if `getDocgenSection` returns `null`:

https://github.com/storybookjs/storybook/blob/9a07fcba8a581558d0fa8262989ef83b70a507a0/code/lib/docs-tools/src/argTypes/docgen/utils/docgenInfo.ts#L14-L16

and then the logic will be handled by `react-element-to-jsx-string`:

https://github.com/algolia/react-element-to-jsx-string/blob/984f26908c17cea134fa53d17ac51f179e5195b2/src/parser/parseReactElement.js#L47-L77

We don't need to rewrite redundant code here.

**Comment by zhyd1997** [comment]
Hi @shilman 

I need your help with why `test-runner-sandboxes` failed on CircleCI, I have no idea about that...

**Comment by ndelangen** [comment]
Having difficulty understanding this ticket & PR, @IanVS @andezzat @shilman can you help? I think if any of you feel confident this is the right thing to do, this can be merged!

**Comment by fizz3r** [comment]
@IanVS, @andezzat @shilman , any update on this PR? Still seeing the issue mentioned in https://github.com/storybookjs/storybook/issues/11554.

**Comment by zhyd1997** [comment]
Hi @ndelangen @shilman @IanVS @kasperpeulen @valentinpalkovic 

Anything I can do on my side to improve the code confidence? 👀

**Review by ndelangen — APPROVED** [review]

**Review by JReinhold — APPROVED** [review]

**Comment by JReinhold** [comment]
I don't know much about this code path TBH, but I like the tests and everything is passing, so I'm okay with merging it.

**Pull request merged**